src/annotator.cpp

```diff
@@ -437,57 +457,71 @@ void Annotator::setModel(const ModelPtr &model)
+    issue->mPimpl->setLevel(Issue::Level::WARNING);
+    issue->mPimpl->setReferenceRule(Issue::ReferenceRule::ANNOTATOR_ID_NOT_FOUND);
     mAnnotator->addIssue(issue);
 }
 
 void Annotator::AnnotatorImpl::addIssueNonUnique(const std::string &id) const
 {
-    auto issue = Issue::create();
-    issue->setDescription("The identifier '" + id + "' occurs " + std::to_string(mIdList.count(id)) + " times in the model so a unique item cannot be located.");
-    issue->setLevel(Issue::Level::WARNING);
-    issue->setReferenceRule(Issue::ReferenceRule::ANNOTATOR_ID_NOT_UNIQUE);
+    auto issue = Issue::IssueImpl::create();
+    issue->mPimpl->setDescription("The identifier '" + id + "' occurs " + std::to_string(mIdList.count(id)) + " times in the model so a unique item cannot be located.");
+    issue->mPimpl->setLevel(Issue::Level::WARNING);
+    issue->mPimpl->setReferenceRule(Issue::ReferenceRule::ANNOTATOR_ID_NOT_UNIQUE);
     mAnnotator->addIssue(issue);
 }
 
 void Annotator::AnnotatorImpl::addIssueNoModel() const
 {
-    auto issue = Issue::create();
-    issue->setDescription("This Annotator object does not have a model to work with.");
-    issue->setLevel(Issue::Level::ERROR);
-    issue->setReferenceRule(Issue::ReferenceRule::ANNOTATOR_NO_MODEL);
+    auto issue = Issue::IssueImpl::create();
+    issue->mPimpl->setDescription("This Annotator object does not have a model to work with.");
+    issue->mPimpl->setReferenceRule(Issue::ReferenceRule::ANNOTATOR_NO_MODEL);
     mAnnotator->addIssue(issue);
 }
 
 void Annotator::AnnotatorImpl::addInvalidArgument(CellmlElementType type) const
 {
-    auto issue = Issue::create();
+    auto issue = Issue::IssueImpl::create();
     auto description = "The item is internally inconsistent: the enum type '" + cellmlElementTypeAsString(type) + "' cannot be used with the stored item.";
-    issue->setDescription(description);
-    issue->setLevel(Issue::Level::ERROR);
-    issue->setReferenceRule(Issue::ReferenceRule::ANNOTATOR_INCONSISTENT_TYPE);
+    issue->mPimpl->setDescription(description);
+    issue->mPimpl->setReferenceRule(Issue::ReferenceRule::ANNOTATOR_INCONSISTENT_TYPE);
     mAnnotator->addIssue(issue);
 }
 
-bool Annotator::AnnotatorImpl::exists(const std::string &id, size_t index) const
+size_t Annotator::AnnotatorImpl::idCount(const std::string &id, size_t index) const
 {
     if (!mAnnotator->hasModel()) {
         addIssueNoModel();
-        return false;
+        return 0;
     }
-    if ((mIdList.count(id) <= index) || (mIdList.count(id) == 0)) {
+
+    auto count = mIdList.count(id);
+    if ((count <= index) || (count == 0)) {
         addIssueNotFound(id);
-        return false;
+        return 0;
     }
-    return true;
+    return count;
 }
 
-AnyItem Annotator::item(const std::string &id)
+bool Annotator::AnnotatorImpl::canReturnItem(const std::string &id, size_t index) const
```

Comment:
its hard to tell with documentation to know what this method should be doing, but the behaviour of `canReturnItem` is different to the previous `exists` method. Not at all clear why an index is even needed if this is now going to error out with non-unique ids.

Reply:
oops, it doesn't error out but return's `true` yet still sets an issue? This behaviour seems weird...

Reply:
I think this is down to the philosophy of the `Annotator`.  The idea is that `id`s should be unique and if they aren't then the user should be told about it.

Reply:
The name `canReturnItem` is more of a hangover from when there was a `std::pair` involved.  I think now it makes more sense to call this method `exists` again.

Reply:
> I think this is down to the philosophy of the Annotator. The idea is that ids should be unique and if they aren't then the user should be told about it.

but this internal API change is not reflected in the current public API documentation or philosophy...or have I missed something?

Reply:
I have made some changes to clarify this.